packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,744 @@
+  }
+
+  /**
+   * Return a CloudWatch Metric indicating job failure.
+   *
+   * This metric is based on the Rule returned by no-args onFailure() call.
+   */
+  public metricFailure(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return JobBase.metricRule(this.onFailure(), props);
+  }
+
+  /**
+   * Return a CloudWatch Metric indicating job timeout.
+   *
+   * This metric is based on the Rule returned by no-args onTimeout() call.
+   */
+  public metricTimeout(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return JobBase.metricRule(this.onTimeout(), props);
+  }
+
+  /**
+   * Creates a new rule for a transition into the input jobState or attempt to create-if-necessary and retrieve the default rule
+   * - A new rule is created but not cached if the id parameter is specified
+   * - A create/retrieve from cache scenario happens when no explicit id (and options) are not provided
+   * The reason is that the default rule is used by onSuccess, onFailure and onTimeout methods which are in turn used by metrics methods.
+   *
+   * @param jobState the job state
+   * @param id optional construct id
+   * @param options optional event options
+   * @private
+   */
+  private rule(jobState: JobState, id?: string, options: events.OnEventOptions = {}): events.Rule {
+    // Caching (for metric methods and default arg-less event methods)
+    const cachedRuleId = `${jobState}Rule`;
+    const cachedRule = this.node.tryFindChild(cachedRuleId);
+    // Use the already created rule if no id is provided (arg-less event methods or events supporting metrics)
+    if (!id && cachedRule) {
+      return cachedRule as events.Rule;
+    }
+    const rule = this.onEvent(id || cachedRuleId, {
+      description: `Rule triggered when Glue job ${this.jobName} is in ${jobState} state`,
+      ...options,
+    });
+    rule.addEventPattern({
+      detail: {
+        state: [jobState],
+      },
+    });
+    return rule;
+  }
+
+}
+
+/**
+ * Properties for enabling Spark UI monitoring feature for Spark-based Glue jobs.
+ *
+ * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+ * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+ */
+export interface SparkUIProps {
+  /**
+   * The bucket where the Glue job stores the logs.
+   *
+   * @default a new bucket will be created.
+   */
+  readonly bucket?: s3.IBucket;
+
+  /**
+   * The path inside the bucket (objects prefix) where the Glue job stores the logs.
+   *
+   * @default - no path will be used, the logs will be written at the root of the bucket.
+   */
+  readonly path?: string;
+}
+
+/**
+ * The Spark UI monitoring configurations for Spark-based Glue jobs.
+ *
+ * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+ * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+ */
+export interface SparkUIConfig {
```

Comment:
shouldn't need to be used outside this file
```suggestion
interface SparkUIConfig {
```

Reply:
SparkUIConfig is used as the type for the `Job.sparkUIConfig` which gives job users access to the created bucket in case one was not explicitly provided.